src/models/APISupport.ts

```diff
@@ -101,28 +97,59 @@ export class APISupport {
+        urlSuffix = this.baseURL+urlSuffix;
+        let header = await this.authContext.getDataPlaneHeader(CallTypeForDP.get);
+        let appComponentsResult = await FetchUtil.httpClientRetries(urlSuffix,header,'get',3,"");
+        if(appComponentsResult.message.statusCode == 200) {
+            let appComponentsObj:AppComponents = await Util.getResultObj(appComponentsResult);
+            for(let guid in appComponentsObj.components){
+                let resourceId = appComponentsObj.components[guid]?.resourceId ?? "";
+                if(this.existingParams.appComponents.has(resourceId?.toLowerCase())) {
+                    let existingGuids = this.existingParams.appComponents.get(resourceId?.toLowerCase()) ?? [];
+                    existingGuids.push(guid);
+                    this.existingParams.appComponents.set(resourceId.toLowerCase(), existingGuids);
+                } else {
+                    this.existingParams.appComponents.set(resourceId.toLowerCase(), [guid]);
+                }
+            }
+        }
+    }
+
+    async getServerMetricsConfig() {
+        let urlSuffix = "tests/"+this.testId+"/server-metrics-config/"+"?api-version="+ ApiVersionConstants.latestVersion;
+        urlSuffix = this.baseURL+urlSuffix;
+        let header = await this.authContext.getDataPlaneHeader(CallTypeForDP.get);
+        let serverComponentsResult = await FetchUtil.httpClientRetries(urlSuffix,header,'get',3,"");
+        if(serverComponentsResult.message.statusCode == 200) {
+            let serverComponentsObj: ServerMetricConfig = await Util.getResultObj(serverComponentsResult);
+            this.yamlModel.mergeExistingServerCriteria(serverComponentsObj);
+        }
+    }
 
     async deleteFileAPI(filename:string) {
-        var urlSuffix = "tests/"+this.testId+"/files/"+filename+"?api-version="+ ApiVersionConstants.tm2024Version;
+        var urlSuffix = "tests/"+this.testId+"/files/"+filename+"?api-version="+ ApiVersionConstants.latestVersion;
         urlSuffix = this.baseURL+urlSuffix;
         let header = await this.authContext.getDataPlaneHeader(CallTypeForDP.delete);
         let delFileResult = await FetchUtil.httpClientRetries(urlSuffix,header,'del',3,"");
         if(delFileResult.message.statusCode != 204) {
-            let delFileObj:any=await Util.getResultObj(delFileResult);
-            let Message: string = delFileObj ? delFileObj.message : Util.ErrorCorrection(delFileResult);
+            let errorObj:any=await Util.getResultObj(delFileResult);
+            let Message: string = errorObj ? errorObj.message : Util.ErrorCorrection(delFileResult);
             throw new Error(Message);
         }
     }
 
     async createTestAPI() {
-        let urlSuffix = "tests/"+this.testId+"?api-version="+ ApiVersionConstants.tm2024Version;
+        let urlSuffix = "tests/"+this.testId+"?api-version="+ ApiVersionConstants.latestVersion;
```

Comment:
For a later PR, we can refactor the API version fetching into a function `getApiVersion()`. That way whenever we want to change the API version, we dont need to change all methods here